metaflow/plugins/tag_cli.py

```diff
@@ -0,0 +1,531 @@
+)
+@click.option(
+    "--remove",
+    "tags_to_remove",
+    multiple=True,
+    default=None,
+    help="Remove this tag to a run. Must specify one or more tags to remove.",
+)
+@click.pass_obj
+def replace_many(obj, run_id, user_namespace, tags_to_add=None, tags_to_remove=None):
+    # While run.replace_tags() can accept 0 additions or 0 removals, we want to encourage
+    # the *obvious* way to achieve their goals. E.g. if they are only adding tags, use "tag add"
+    # over more obscure "tag replace --add ... --add ..."
+    if not tags_to_add and not tags_to_remove:
+        raise CommandException(
+            "Specify at least one tag to add (--add) and one tag to remove (--remove)"
+        )
+    if not tags_to_remove:
+        raise CommandException(
+            "Specify at least one tag to remove; else please use *tag add*."
+        )
+    if not tags_to_add:
+        raise CommandException(
+            "Specify at least one tag to add, else please use *tag remove*."
+        )
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    run = _get_client_run_obj(obj, run_id, user_namespace)
+
+    run.replace_tags(tags_to_remove, tags_to_add)
+
+    obj.echo("Operation successful. New tags:")
+    _print_tags_for_one_run(obj, run)
+
+
+@tag.command("list", help="List tags of a run.")
+@click.option(
+    "--run-id",
+    required=False,
+    default=None,
+    type=str,
+    help="Run ID of the specific run to list.",
+)
+@click.option(
+    "--all",
+    "list_all",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="List tags across all runs of this flow.",
+)
+@click.option(
+    "--my-runs",
+    "my_runs",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="List tags across all runs of the flow under the default namespace.",
+)
+@click.option(
+    "--hide-system-tags",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Hide system tags.",
+)
+@click.option(
+    "--by-tag",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Display results by showing runs grouped by tags",
+)
+@click.option(
+    "--by-run",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Display tags grouped by run",
+)
+@click.argument("arg_run_id", required=False, default=None, type=str)
+@click.pass_obj
+def tag_list(
+    obj, run_id, hide_system_tags, list_all, my_runs, by_tag, by_run, arg_run_id
+):
+    if run_id is None and arg_run_id is None and not list_all and not my_runs:
+        # Assume list_all by default
+        list_all = True
+
+    if list_all and my_runs:
+        raise CommandException(
+            "Option --all cannot be used together with --my-runs.\n"
+            "You can run the flow with commands 'tag list --help' "
+            "to check the meaning of the two options."
+        )
+
+    if run_id is not None and arg_run_id is not None:
```

Comment:
Is `arg_run_id` for nflx backwards compatibility? If so, can we notate that fact in the code?

Reply:
cc @romain-intel re: `arg_run_id` - should we keep it and add a comment, or can we remove it?

Reply:
We do have this currently so yes, it would be for backward compatibility. It seems easy enough to keep so would prefer if possible.